### ISSUE TITLE
Malformed switch expression syntax causes VS to crash to desktop when NRTs are enabled

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn#45862]
**Version Used**: VS 16.6.3, 16.7-p3.1

It repros every time. VS freezes for a couple of seconds and then crashes to the desktop.

Paste this (simplified) code into the editor:

```cs
#nullable enable

class C
{
    void M()
    {
        _ = 0 switch
        {
            0 =
            _ = null,
        };
    }
}
```

Repros as the only file with this .csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
  </PropertyGroup>

</Project>
```

**Comment by jnm2** [comment]
Verified that this takes down 16.6.3 the same way.

**Comment by kevinsun-dev** [comment]
Fixed in PR #45974. Thanks @jnm2 for reporting the issue!
